lib/asset_checker.rb

```diff
@@ -0,0 +1,39 @@
+class AssetChecker
+  def self.run(argv)
+    assets_file = 'app/assets/javascripts/assets.js.erb'
+    translations_file = 'app/assets/javascripts/i18n-strings.js.erb'
+    @asset_strings = load_included_strings(assets_file)
+    @translation_strings = load_included_strings(translations_file)
+    argv.map { |f| check_file(f) }
+  end
+
+  def self.check_file(file)
+    data = File.open(file).read
+    missing_translations = find_missing(data, /\Wt\(['"](.*)['"]\)/, @translation_strings)
+    missing_assets = find_missing(data, /\WassetPath=["'](.*)['"]/, @asset_strings)
+    if missing_translations.any? || missing_assets.any? # rubocop:disable Style/GuardClause
+      warn file
+      missing_translations.each do |t|
+        warn "Missing translation, #{t}"
+      end
+      missing_assets.each do |a|
+        warn "Missing asset, #{a}"
+      end
```

Comment:
Depending how this checker could be integrated into e.g. continuous integration, might it be worth raising an exception, or otherwise ensuring that the script exits with a non-zero exit code, so that it can be treated as a failure?

Reply:
Added 👍